### PR TITLE
SCT-511 - App redirects to login when "Clear search" is used

### PR DESCRIPTION
**Pull request by mapleleafjack**
**What**  
Added  check for data != 1 (useSWRinfinite)

**Why**  
When searching for residents, and using the "Clear search" mechanism, the app redirects to the login page due to an error

This is not the definitive solution, but it will prevents the app from breaking when the user search twice

### CODE REVIEW

**Review by chubberlisk — COMMENTED** [review]
Looks like we have [Cypress tests that cover searching](https://github.com/LBHackney-IT/lbh-social-care-frontend/blob/c0fef49940f698c23da93e1ef417e97cb85130aa/cypress/integration/search_for_person.ts#L3), could we add tests to cover this?

File: components/Search/Search.jsx [hunk 1]
```diff
@@ -142,7 +143,7 @@ const Search = ({
           )
         )}
       </div>
-      {error && <ErrorMessage />}
+      {error && data != 1 && <ErrorMessage />}
```
Comment (chrishutchinson) [inline components/Search/Search.jsx:146]: What happens if for some reason there is an error (say a HTTP error) and the value of `data` is `1`? Is that scenario possible?

File: components/Search/Search.jsx [hunk 2]
```diff
@@ -67,10 +67,11 @@ const Search = ({
     query,
     hasQuery || showOnlyMyResults
   );
-  const results = data && {
-    records: data?.reduce((acc, d) => [...acc, ...getRecords(d)], []),
-    nextCursor: data[data.length - 1].nextCursor,
-  };
+  const results = data &&
+    data != 1 && {
```
Comment (chrishutchinson) [inline components/Search/Search.jsx:71]: Probably best to use strict (in)equality: `data !== 1`. Same in your change further down too

File: components/Search/Search.jsx (hunk 1 above)
Comment (mapleleafjack) [reply components/Search/Search.jsx:146]: It should be not possible, i've  tested it with a "wrong" connection string and it returns an empty array - triggering the error correctly

File: components/Search/Search.jsx (hunk 2 above)
Comment (mapleleafjack) [reply components/Search/Search.jsx:71]: Done ;)

**Comment by jhackett1** [comment]
what would the definitive fix be?

**Review by jhackett1 — APPROVED** [review]

**Comment by mapleleafjack** [comment]
> what would the definitive fix be?

Hi @jhackett1,

The definitie fix would be the one in SCT-610

**Pull request merged**